adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenComponentController.java

```diff
@@ -0,0 +1,205 @@
+/*
+ *                        ######
+ *                        ######
+ *  ############    ####( ######  #####. ######  ############   ############
+ *  #############  #####( ######  #####. ######  #############  #############
+ *         ######  #####( ######  #####. ######  #####  ######  #####  ######
+ *  ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ *  ###### ######  #####( ######  #####. ######  #####          #####  ######
+ *  #############  #############  #############  #############  #####  ######
+ *   ############   ############  #############   ############  #####  ######
+ *                                       ######
+ *                                #############
+ *                                ############
+ *
+ *  Adyen Hybris Extension
+ *
+ *  Copyright (c) 2020 Adyen B.V.
+ *  This file is open source and available under the MIT license.
+ *  See the LICENSE file for more info.
+ */
+package com.adyen.v6.controllers.pages;
+
+import com.adyen.model.checkout.PaymentsResponse;
+import com.adyen.model.checkout.details.PayPalDetails;
+import com.adyen.service.exception.ApiException;
+import com.adyen.v6.exceptions.AdyenComponentException;
+import com.adyen.v6.exceptions.AdyenNonAuthorizedPaymentException;
+import com.adyen.v6.facades.AdyenCheckoutFacade;
+import com.adyen.v6.forms.AdyenPaymentForm;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import de.hybris.platform.acceleratorfacades.flow.CheckoutFlowFacade;
+import de.hybris.platform.acceleratorfacades.order.AcceleratorCheckoutFacade;
+import de.hybris.platform.commercefacades.order.data.CartData;
+import de.hybris.platform.order.InvalidCartException;
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import static com.adyen.v6.constants.AdyenControllerConstants.COMPONENT_PREFIX;
+
+@Controller
+@RequestMapping(COMPONENT_PREFIX)
+public class AdyenComponentController {
```

Comment:
is this controller meant to be used only for PayPal? or is it meant to be used by other methods in the future?

Reply:
On the story we decided to try to make things generic, so it could be used by other methods, so I used a generic name.
The implementation is still specific for Paypal though, because of limitations of our library right now (but when we create the helper for the `details` classes, I think it can be made more generic)

Reply:
It is generic and may be used for others in future.